crates/stackable-telemetry/src/instrumentation/axum/mod.rs

```diff
@@ -375,25 +443,22 @@ impl SpanExt for Span {
             span.record("client.address", client_socket_address.ip().to_string());
 
             if opt_in {
-                span.record("client.port", client_socket_address.port());
+                // NOTE (@Techassi): We cast to i64, because otherwise the field
+                // will NOT be recorded as a number but as a string. This is
+                // likely an issue in the tracing-opentelemetry crate.
+                span.record("client.port", client_socket_address.port() as i64);
```

Comment:
**note (non-blocker)**: IMO, we should get rid of the `opt-in` concept here, and just decide on the fields we record.

The whole opt-in thing was more for implementers (in this case, us) to choose which of those they want to add to instrumentation.

Reply:
Sure, I can remove the opt-in feature. In the future, this might become a problem tho. If we can support recording the HTTP request headers as fields, this can result in a large set of additional fields.

We can also remove it for now and decide later, if we need such a mechanism and in which form.

Reply:
Added quotes in d41179b

Reply:
> If we can support recording the HTTP request headers as fields, this can result in a large set of additional fields.

We could make that decision on whether we want to record such detail. Also, there are protections to limit the amount of attributes that can be sent in the span.

Nothing needed right now, we can remove it later (if we want).